lib/Spot/Query.php

```diff
@@ -651,32 +666,63 @@ public function escape($string)
     }
 
     /**
-     * Escape/quote direct user input
+     * Escape/quote multiple identifiers
      *
-     * @param string $field
+     * @param array $identifiers
+     * @return array
+     */
+    protected function escapeMultipleIdentifiers(array $identifiers) {
+        array_walk($identifiers, function (&$identifier) {
+            $identifier = $this->escapeIdentifier($identifier);
+        });
+
+        return $identifiers;
+    }
+
+    /**
+     * Removes escape/quote character
+     *
+     * @param string $identifier
      * @return string
-     * @throws Exception
      */
-    public function escapeField($field)
+    public function unescapeIdentifier($identifier)
     {
-        if ($this->_noQuote) {
-            return $field;
+        if (strpos($identifier, ".") !== false) {
+            $parts = array_map(array($this, "unescapeIdentifier"), explode(".", $identifier));
+
+            return implode(".", $parts);
         }
 
-        return $this->mapper()->connection()->quoteIdentifier($field);
+        return trim($identifier, $this->mapper()->connection()->getDatabasePlatform()->getIdentifierQuoteCharacter());
+    }
+
+    /**
+     * Escape/quote identifier
+     *
+     * @param string $identifier
+     * @return string
+     */
+    public function escapeIdentifier($identifier)
```

Comment:
I understand this and I still think this is a bad idea.

If I create one function that casts every passed identifier to array and handles as array - then in `fieldWithAlias` I need to get the first element of returned array - which isn't a good idea.
Or, I create a function that handles both - single and multiple identifiers and returns accordingly - then I have code duplication that needs to be extracted to separate method. But escaping method has inconsistient return types.

Or I create two methods - one for single identifier and one for array of identifiers - like I did.

If you still insist - then I will merge them into one function.